### PR TITLE
fft

### PR DESCRIPTION
**Pull request by gurumoorty-s**

### CODE REVIEW

**Comment by Guru-mp3** [comment]
Hey @AnnaTz are there any changes that are required?? I checked that there are no formatting errors but there's run_tests_1 which is failing, can you help me out??

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    fn_tree="jax.numpy.fft.fft",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float"),
+                                         array_api_dtypes=True,
+                                         shape=helpers.ints(min_value=2).
+                                         map(lambda x: tuple([x, ])))
+    .filter(
+        lambda x: "bfloat16" not in x[0]
+        and "float16" not in x[0]
+    )
+)
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py:17]: You will need to generate and pass all function arguments, including `a`, `n`, `axis` and `norm`.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+
+
+# fft
+@handle_frontend_test(
+    fn_tree="jax.numpy.fft.fft",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float"),
+                                         array_api_dtypes=True,
+                                         shape=helpers.ints(min_value=2).
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py:11]: I don't see any limitation on the shape of the array mentioned in the documentation. So, I think you shouldn't set the `shape` argument at all.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# local
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+
+
+# fft
+@handle_frontend_test(
+    fn_tree="jax.numpy.fft.fft",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float"),
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py:9]: This should be `get_dtypes("valid")`. If you get errors for any dtype then you will need to add the `@with_unsupported_dtypes` decorator to the function as described here https://lets-unify.ai/ivy/overview/deep_dive/data_types.html#supported-and-unsupported-data-types

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+@handle_frontend_test(
+    fn_tree="jax.numpy.fft.fft",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("float"),
+                                         array_api_dtypes=True,
+                                         shape=helpers.ints(min_value=2).
+                                         map(lambda x: tuple([x, ])))
+    .filter(
+        lambda x: "bfloat16" not in x[0]
+        and "float16" not in x[0]
+    )
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py:16]: Along with changing the available dtypes to "valid", this `filter` should be removed.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by Guru-mp3** [comment]
Hello @AnnaTz , thanks for reviewing my PR, I have added changes needed in the files. One thing that I found out while testing is that without the shape parameter, the `helper.dtype_and_values` function was generating values which had no shape, even after conversion from IvyArray to Native Array, so I needed to add the shape parameter, please check, if I need to do further changes.

File: ivy/functional/frontends/jax/numpy/fft.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# local
+import ivy
+from ivy.func_wrapper import with_unsupported_dtypes
+from ivy.functional.frontends.jax.func_wrapper import to_ivy_arrays_and_back
+
+
+@to_ivy_arrays_and_back
+@with_unsupported_dtypes({"0.4.8 and below": ("bool",)}, "jax")
+def fft(a, n=None, dim=-1, norm=None):
```
Comment (AnnaTz) [inline ivy/functional/frontends/jax/numpy/fft.py:9]: The signature of a frontend function should be exactly the same as the original one. I think `dim` should be named `axis`.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+# local
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+
+
+# fft
+@handle_frontend_test(
+    fn_tree="jax.numpy.fft.fft",
+    dtype_and_x=helpers.dtype_and_values(available_dtypes=helpers.get_dtypes("valid"),
+                                           shape=helpers.ints(min_value=0).
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py:13]: To avoid getting an empty shape, the `min_num_dims` can be set. No need to set and map the `shape` argument.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+                                           map(lambda x: tuple([x, ]))),
+    dim=helpers.ints(min_value=-1, max_value=0),
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_fft.py:15]: The generation of the axis should depend on the input array's shape and size, otherwise the axis might be invalid. You could use `dtype_values_axis` to get correct array-axis pairs.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
